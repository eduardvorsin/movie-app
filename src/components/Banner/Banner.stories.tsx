import type { Meta, StoryObj } from '@storybook/react';
import Banner from './Banner';
import Button from '../Button/Button';

const meta: Meta<typeof Banner> = {
	title: 'components/Banner',
	tags: ['autodocs'],
	argTypes: {
		appearance: {
			description: 'Banner appearance',
		},
		className: {
			description: 'Additional classes for the banner'
		},
		children: {
			description: 'Banner content',
		},
		testId: {
			description: 'Id for testing the component',
		},
		title: {
			description: 'Banner title',
		},
		hideIcon: {
			description: 'Show or hide the icon in front of the content',
		},
		onClose: {
			control: false,
			description: 'A callback that is triggered when the close button is clicked',
		},
		closeButton: {
			description: 'Show or hide the banner close button',
		},
		titleLevel: {
			description: 'Header level from 1 to 6',
		},
		actions: {
			control: false,
			description: 'Array of action buttons for the banner',
		}
	},
	component: Banner,
	parameters: {
		docs: {
			description: {
				component: 'This component is utilized to apprise merchants of significant alterations or ongoing situations. Employ this element when you want to convey important messages to merchants in a noticeable manner. Banners are positioned at the upper part of the relevant page or section, just beneath the page or section heading.',
			},
		},
	},
}

export default meta;
type Story = StoryObj<typeof Banner>;

export const Info: Story = {
	args: {
		appearance: 'info',
		title: 'Info',
		children: 'Info banner Lorem ipsum dolor sit amet consectetur adipisicing elit. Nesciunt non repellendus ullam quam dolore inventore deleniti reiciendis esse dolor sed.'
	},
	parameters: {
		docs: {
			description: {
				story: 'The banner is the default presentation utilized to indicate a shift in status or convey vital information.',
			},
		},
	},
};

export const Success: Story = {
	args: {
		appearance: 'success',
		title: 'Success',
		children: 'Success banner Lorem ipsum dolor sit amet consectetur adipisicing elit. Nesciunt non repellendus ullam quam dolore inventore deleniti reiciendis esse dolor sed.'
	},
	parameters: {
		docs: {
			description: {
				story: 'Utilize a success banner message to inform the user that an action or event has been successfully completed.',
			},
		},
	},
};

export const Warning: Story = {
	args: {
		appearance: 'warning',
		title: 'Warning',
		children: 'Warning banner Lorem ipsum dolor sit amet consectetur adipisicing elit. Nesciunt non repellendus ullam quam dolore inventore deleniti reiciendis esse dolor sed.'
	},
	parameters: {
		docs: {
			description: {
				story: 'Use a warning banner message to help users avoid errors, manage authentication issues, take actions to prevent potentially dangerous actions, and feel confident in their decision-making, such as in confirmation modals.',
			},
		},
	},
};

export const Danger: Story = {
	args: {
		appearance: 'danger',
		title: 'Danger',
		children: 'Danger banner Lorem ipsum dolor sit amet consectetur adipisicing elit. Nesciunt non repellendus ullam quam dolore inventore deleniti reiciendis esse dolor sed.'
	},
	parameters: {
		docs: {
			description: {
				story: 'Use an error banner message to communicate when something destructive or critical has occurred, access has been denied, or connectivity issues are present. It\'s worth noting that employing section messages for these scenarios is not a common practice.',
			},
		},
	},
};

export const Discovery: Story = {
	args: {
		appearance: 'discovery',
		title: 'Discovery',
		children: 'Discovery banner Lorem ipsum dolor sit amet consectetur adipisicing elit. Nesciunt non repellendus ullam quam dolore inventore deleniti reiciendis esse dolor sed.'
	},
	parameters: {
		docs: {
			description: {
				story: 'Utilize a discovery banner message to indicate a UI update or deliver information regarding new features and the onboarding process.',
			},
		},
	},
};

export const CloseButton: Story = {
	argTypes: {
		onClose: {
			action: 'Closed',
		},
	},
	args: {
		appearance: 'info',
		title: 'Close Button',
		closeButton: true,
		children: 'Close button banner Lorem ipsum dolor sit amet consectetur adipisicing elit. Nesciunt non repellendus ullam quam dolore inventore deleniti reiciendis esse dolor sed.'
	},
	parameters: {
		docs: {
			description: {
				story: 'Adds a button with which this banner can be hidden',
			},
		},
	},
};

export const HideIcon: Story = {
	args: {
		appearance: 'success',
		title: 'Hide Icon',
		hideIcon: true,
		children: 'Hide icon banner Lorem ipsum dolor sit amet consectetur adipisicing elit. Nesciunt non repellendus ullam quam dolore inventore deleniti reiciendis esse dolor sed.'
	},
	parameters: {
		docs: {
			description: {
				story: 'Hides the icon to the left of the content',
			},
		},
	},
};

export const WithoutTitle: Story = {
	args: {
		children: 'Without Title banner Lorem ipsum dolor sit amet consectetur adipisicing elit. Nesciunt non repellendus ullam quam dolore inventore deleniti reiciendis esse dolor sed.'
	},
	parameters: {
		docs: {
			description: {
				story: 'If you do not pass the value to the property for the header, it will not be displayed in the banner',
			},
		},
	},
};

export const WithActions: Story = {
	args: {
		appearance: 'discovery',
		title: 'With Actions',
		children: 'With actions banner Lorem ipsum dolor sit amet consectetur adipisicing elit. Nesciunt non repellendus ullam quam dolore inventore deleniti reiciendis esse dolor sed.',
		actions: [
			<Button appearance='discovery'>Add</Button>,
			<Button appearance='discovery'>Remove</Button>
		]
	},
	parameters: {
		docs: {
			description: {
				story: 'Allows you to add your own buttons or links inside the banner',
			},
		},
	},
};