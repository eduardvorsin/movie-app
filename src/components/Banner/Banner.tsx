'use client';
import { ReactNode } from 'react';
import Warning from '../../assets/icons/warning.svg?url';
import Error from '../../assets/icons/error.svg?url';
import Confirmation from '../../assets/icons/confirmation.svg?url';
import Info from '../../assets/icons/info.svg?url';
import Question from '../../assets/icons/question.svg?url';
import Cancel from '../../assets/icons/cancel.svg?url';
import Button from '../Button/Button';

type Props = {
	className?: string,
	title?: string,
	hideIcon?: boolean,
	children: ReactNode,
	appearance?: 'success' | 'info' | 'warning' | 'danger' | 'discovery',
	onClose?: () => void,
	closeButton?: boolean,
	titleLevel: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6',
	actions?: ReactNode[],
	testId?: string,
};

const appearances = {
	success: 'bg-green-100 dark:bg-green-1000',
	info: 'bg-blue-100 dark:bg-blue-1000',
	warning: 'bg-yellow-100 dark:bg-yellow-1000',
	danger: 'bg-red-100 dark:bg-red-1000',
	discovery: 'bg-purple-100 dark:bg-purple-1000',
} as const;

const apperanceIcons = {
	discovery: (
		<svg className='fill-purple-700 dark:fill-purple-400 ' viewBox='0 0 20 20'>
			<use href={`${Question.src}#question`}></use>
		</svg>
	),
	success: (
		<svg className='fill-green-800 dark:fill-green-400' viewBox='0 0 20 20'>
			<use href={`${Confirmation.src}#confirmation`}></use>
		</svg>
	),
	info: (
		<svg className='fill-blue-700 dark:fill-blue-400' viewBox='0 0 20 20'>
			<use href={`${Info.src}#info`}></use>
		</svg>
	),
	warning: (
		<svg className='fill-yellow-600 dark:fill-yellow-500' viewBox='0 0 20 20'>
			<use href={`${Warning.src}#warning`}></use>
		</svg>
	),
	danger: (
		<svg className='fill-red-700 dark:fill-red-400' viewBox='0 0 20 20'>
			<use href={`${Error.src}#error`}></use>
		</svg>
	),
} as const;

export default function Banner({
	appearance = 'info',
	className,
	title,
	hideIcon,
	children,
	actions,
	onClose,
	closeButton,
	titleLevel,
	testId,
}: Props) {
	const classes = [
		'py-4 pl-4 rounded-[0.1875rem] flex relative',
		appearances[appearance],
		closeButton ? 'pr-12' : 'pr-4',
		className
	].join(' ');
	const Heading = titleLevel;

	return (
		<div
			className={`${classes}`}
			data-testid={testId}
		>
			{!hideIcon && (
				<span
					className='w-6 h-6 mr-4'
				>
					{apperanceIcons[appearance]}
				</span>
			)}
			<div>
				{title && (
					<Heading
						className='mb-2 text-200 font-bold text-neutral-1000 dark:text-dark-neutral-900'
					>
						{title}
					</Heading>
				)}
				<p
					className={'text-100 text-neutral-1000 dark:text-dark-neutral-900'}
				>
					{children}
				</p>
				{actions && (
					<div
						className={'flex mt-3'}
					>
						{actions.map((action, index) => (
							<div key={index} className='mr-2 last:mr-0'>{action}</div>
						))}
					</div>
				)}
			</div>
			{closeButton && (
				<Button
					appearance={appearance === 'info' ? 'primary' : appearance}
					className='text-[0rem] p-2 w-10 h-10 absolute top-2 right-0'
					iconButton
					onClick={onClose}
				>
					close
					<svg className='fill-current w-6 h-6' viewBox='0 0 20 20' aria-hidden>
						<use href={`${Cancel.src}#cancel`}></use>
					</svg>
				</Button>
			)}
		</div>
	)
};
