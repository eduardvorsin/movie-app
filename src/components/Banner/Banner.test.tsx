import { render, screen } from "@testing-library/react";
import Banner from "./Banner";
import userEvent from "@testing-library/user-event";
import Button from "@/components/Button/Button";

describe('Banner tests', () => {
	it('This is rendered correctly', () => {
		render(
			<Banner
				closeButton={false}
				testId='test-banner'
			>
				Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatibus tempora provident accusamus quaerat iure atque labore odit voluptatem quisquam, maiores voluptas veritatis voluptate sed magni quod enim nobis tempore corrupti.
			</Banner>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-banner')).toBeInTheDocument();
	});

	it('if the title prop is empty then the title will not be rendered', () => {
		render(
			<Banner
				closeButton={false}
				testId='test-banner'
			>
				Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatibus tempora provident accusamus quaerat iure atque labore odit voluptatem quisquam, maiores voluptas veritatis voluptate sed magni quod enim nobis tempore corrupti.
			</Banner>
		);

		expect(screen.queryByRole<HTMLHeadingElement>('heading')).not.toBeInTheDocument();
	});


	it('if hideIcon is true, the icon to the left of the text is hidden', () => {
		render(
			<Banner
				closeButton={false}
				hideIcon
				testId='test-banner'
			>
				Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatibus tempora provident accusamus quaerat iure atque labore odit voluptatem quisquam, maiores voluptas veritatis voluptate sed magni quod enim nobis tempore corrupti.
			</Banner>
		);

		expect(screen.queryByRole<HTMLSpanElement>('presentation')).not.toBeInTheDocument();
	});

	it('if the close Button is set to true, the close button of the banner is displayed.', () => {
		render(
			<Banner
				closeButton
				testId='test-banner'
				dictionary={{ closeButton: 'To close' }}
			>
				Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatibus tempora provident accusamus quaerat iure atque labore odit voluptatem quisquam, maiores voluptas veritatis voluptate sed magni quod enim nobis tempore corrupti.
			</Banner>
		);

		expect(screen.getByRole<HTMLButtonElement>('button', { name: /close/i })).toBeInTheDocument();
	});

	it('when you click on the banner close button, the mock function is triggered', async () => {
		const user = userEvent.setup();
		const mockFn = jest.fn();
		render(
			<Banner
				closeButton
				onClose={mockFn}
				testId='test-banner'
				dictionary={{ closeButton: 'To close' }}
			>
				Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatibus tempora provident accusamus quaerat iure atque labore odit voluptatem quisquam, maiores voluptas veritatis voluptate sed magni quod enim nobis tempore corrupti.
			</Banner>
		);

		await user.click(screen.getByRole<HTMLButtonElement>('button', { name: /close/i }))

		expect(mockFn).toHaveBeenCalledTimes(1);
	});


	it('is a basic snapshot', () => {
		render(
			<Banner
				closeButton={false}
				testId='test-banner'
			>
				Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatibus tempora provident accusamus quaerat iure atque labore odit voluptatem quisquam, maiores voluptas veritatis voluptate sed magni quod enim nobis tempore corrupti.
			</Banner>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-banner')).toMatchSnapshot();
	});

	it('is a snapshot with closeButton equal to true', () => {
		render(
			<Banner
				testId='test-banner'
				closeButton
				dictionary={{ closeButton: 'To close' }}
			>
				Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatibus tempora provident accusamus quaerat iure atque labore odit voluptatem quisquam, maiores voluptas veritatis voluptate sed magni quod enim nobis tempore corrupti.
			</Banner>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-banner')).toMatchSnapshot();
	});

	it('is a snapshot with a "success" type appearance', () => {
		render(
			<Banner
				testId='test-banner'
				appearance='success'
				closeButton={false}
			>
				Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatibus tempora provident accusamus quaerat iure atque labore odit voluptatem quisquam, maiores voluptas veritatis voluptate sed magni quod enim nobis tempore corrupti.
			</Banner>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-banner')).toMatchSnapshot();
	});

	it('is a snapshot with a "info" type appearance', () => {
		render(
			<Banner
				testId='test-banner'
				appearance='info'
				closeButton={false}
			>
				Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatibus tempora provident accusamus quaerat iure atque labore odit voluptatem quisquam, maiores voluptas veritatis voluptate sed magni quod enim nobis tempore corrupti.
			</Banner>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-banner')).toMatchSnapshot();
	});

	it('is a snapshot with a "warning" type appearance', () => {
		render(
			<Banner
				testId='test-banner'
				appearance='warning'
				closeButton={false}
			>
				Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatibus tempora provident accusamus quaerat iure atque labore odit voluptatem quisquam, maiores voluptas veritatis voluptate sed magni quod enim nobis tempore corrupti.
			</Banner>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-banner')).toMatchSnapshot();
	});

	it('is a snapshot with a "danger" type appearance', () => {
		render(
			<Banner
				testId='test-banner'
				appearance='danger'
				closeButton={false}
			>
				Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatibus tempora provident accusamus quaerat iure atque labore odit voluptatem quisquam, maiores voluptas veritatis voluptate sed magni quod enim nobis tempore corrupti.
			</Banner>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-banner')).toMatchSnapshot();
	});

	it('is a snapshot with a "discovery" type appearance', () => {
		render(
			<Banner
				testId='test-banner'
				appearance='discovery'
				closeButton={false}
			>
				Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatibus tempora provident accusamus quaerat iure atque labore odit voluptatem quisquam, maiores voluptas veritatis voluptate sed magni quod enim nobis tempore corrupti.
			</Banner>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-banner')).toMatchSnapshot();
	});

	it('is a snapshot with hideIcon equal to true', () => {
		render(
			<Banner
				testId='test-banner'
				hideIcon
				closeButton={false}
			>
				Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatibus tempora provident accusamus quaerat iure atque labore odit voluptatem quisquam, maiores voluptas veritatis voluptate sed magni quod enim nobis tempore corrupti.
			</Banner>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-banner')).toMatchSnapshot();
	});

	it('is a snapshot with action buttons', () => {
		render(
			<Banner
				testId='test-banner'
				closeButton={false}
				actions={[
					<Button key={0}>action</Button>,
					<Button key={1}>action</Button>,
					<Button key={2}> action</Button>,
				]}
			>
				Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatibus tempora provident accusamus quaerat iure atque labore odit voluptatem quisquam, maiores voluptas veritatis voluptate sed magni quod enim nobis tempore corrupti.
			</Banner>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-banner')).toMatchSnapshot();
	});
});