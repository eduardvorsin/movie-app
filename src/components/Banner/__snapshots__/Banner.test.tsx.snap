// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner tests is a basic snapshot 1`] = `
<div
  class="py-4 pl-4 rounded-[0.1875rem] flex relative bg-blue-100 dark:bg-blue-1000 pr-4 "
  data-testid="test-banner"
>
  <span
    class="w-6 h-6 mr-4"
    role="presentation"
  >
    <svg
      class="fill-blue-700 dark:fill-blue-400 transition-colors duration-150"
      viewBox="0 0 20 20"
    >
      <use
        href="/assets/icons/info.svg#info"
      />
    </svg>
  </span>
  <div>
    <p
      class="text-100 text-neutral-1000 dark:text-dark-neutral-900 transition-colors duration-150"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatibus tempora provident accusamus quaerat iure atque labore odit voluptatem quisquam, maiores voluptas veritatis voluptate sed magni quod enim nobis tempore corrupti.
    </p>
  </div>
</div>
`;

exports[`Banner tests is a snapshot with a "danger" type appearance 1`] = `
<div
  class="py-4 pl-4 rounded-[0.1875rem] flex relative bg-red-100 dark:bg-red-1000 pr-4 "
  data-testid="test-banner"
>
  <span
    class="w-6 h-6 mr-4"
    role="presentation"
  >
    <svg
      class="fill-red-700 dark:fill-red-400 transition-colors duration-150"
      viewBox="0 0 20 20"
    >
      <use
        href="/assets/icons/error.svg#error"
      />
    </svg>
  </span>
  <div>
    <p
      class="text-100 text-neutral-1000 dark:text-dark-neutral-900 transition-colors duration-150"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatibus tempora provident accusamus quaerat iure atque labore odit voluptatem quisquam, maiores voluptas veritatis voluptate sed magni quod enim nobis tempore corrupti.
    </p>
  </div>
</div>
`;

exports[`Banner tests is a snapshot with a "discovery" type appearance 1`] = `
<div
  class="py-4 pl-4 rounded-[0.1875rem] flex relative bg-purple-100 dark:bg-purple-1000 pr-4 "
  data-testid="test-banner"
>
  <span
    class="w-6 h-6 mr-4"
    role="presentation"
  >
    <svg
      class="fill-purple-700 dark:fill-purple-400 transition-colors duration-150"
      viewBox="0 0 20 20"
    >
      <use
        href="/assets/icons/question.svg#question"
      />
    </svg>
  </span>
  <div>
    <p
      class="text-100 text-neutral-1000 dark:text-dark-neutral-900 transition-colors duration-150"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatibus tempora provident accusamus quaerat iure atque labore odit voluptatem quisquam, maiores voluptas veritatis voluptate sed magni quod enim nobis tempore corrupti.
    </p>
  </div>
</div>
`;

exports[`Banner tests is a snapshot with a "info" type appearance 1`] = `
<div
  class="py-4 pl-4 rounded-[0.1875rem] flex relative bg-blue-100 dark:bg-blue-1000 pr-4 "
  data-testid="test-banner"
>
  <span
    class="w-6 h-6 mr-4"
    role="presentation"
  >
    <svg
      class="fill-blue-700 dark:fill-blue-400 transition-colors duration-150"
      viewBox="0 0 20 20"
    >
      <use
        href="/assets/icons/info.svg#info"
      />
    </svg>
  </span>
  <div>
    <p
      class="text-100 text-neutral-1000 dark:text-dark-neutral-900 transition-colors duration-150"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatibus tempora provident accusamus quaerat iure atque labore odit voluptatem quisquam, maiores voluptas veritatis voluptate sed magni quod enim nobis tempore corrupti.
    </p>
  </div>
</div>
`;

exports[`Banner tests is a snapshot with a "success" type appearance 1`] = `
<div
  class="py-4 pl-4 rounded-[0.1875rem] flex relative bg-green-100 dark:bg-green-1000 pr-4 "
  data-testid="test-banner"
>
  <span
    class="w-6 h-6 mr-4"
    role="presentation"
  >
    <svg
      class="fill-green-800 dark:fill-green-400 transition-colors duration-150"
      viewBox="0 0 20 20"
    >
      <use
        href="/assets/icons/confirmation.svg#confirmation"
      />
    </svg>
  </span>
  <div>
    <p
      class="text-100 text-neutral-1000 dark:text-dark-neutral-900 transition-colors duration-150"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatibus tempora provident accusamus quaerat iure atque labore odit voluptatem quisquam, maiores voluptas veritatis voluptate sed magni quod enim nobis tempore corrupti.
    </p>
  </div>
</div>
`;

exports[`Banner tests is a snapshot with a "warning" type appearance 1`] = `
<div
  class="py-4 pl-4 rounded-[0.1875rem] flex relative bg-yellow-100 dark:bg-yellow-1000 pr-4 "
  data-testid="test-banner"
>
  <span
    class="w-6 h-6 mr-4"
    role="presentation"
  >
    <svg
      class="fill-yellow-600 dark:fill-yellow-500 transition-colors duration-150"
      viewBox="0 0 20 20"
    >
      <use
        href="/assets/icons/warning.svg#warning"
      />
    </svg>
  </span>
  <div>
    <p
      class="text-100 text-neutral-1000 dark:text-dark-neutral-900 transition-colors duration-150"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatibus tempora provident accusamus quaerat iure atque labore odit voluptatem quisquam, maiores voluptas veritatis voluptate sed magni quod enim nobis tempore corrupti.
    </p>
  </div>
</div>
`;

exports[`Banner tests is a snapshot with action buttons 1`] = `
<div
  class="py-4 pl-4 rounded-[0.1875rem] flex relative bg-blue-100 dark:bg-blue-1000 pr-4 "
  data-testid="test-banner"
>
  <span
    class="w-6 h-6 mr-4"
    role="presentation"
  >
    <svg
      class="fill-blue-700 dark:fill-blue-400 transition-colors duration-150"
      viewBox="0 0 20 20"
    >
      <use
        href="/assets/icons/info.svg#info"
      />
    </svg>
  </span>
  <div>
    <p
      class="text-100 text-neutral-1000 dark:text-dark-neutral-900 transition-colors duration-150"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatibus tempora provident accusamus quaerat iure atque labore odit voluptatem quisquam, maiores voluptas veritatis voluptate sed magni quod enim nobis tempore corrupti.
    </p>
    <div
      class="flex mt-3"
    >
      <div
        class="mr-2 last:mr-0"
      >
        <button
          class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-100  bg-blue-700 text-neutral-0 dark:text-dark-neutral-0 enabled:hover:bg-blue-800 enabled:active:bg-blue-900 dark:bg-blue-400 dark:enabled:hover:bg-blue-300 dark:enabled:active:bg-blue-200 py-2 px-4 relative  "
        >
          <span
            class="inline-flex items-center"
          >
            action
          </span>
        </button>
      </div>
      <div
        class="mr-2 last:mr-0"
      >
        <button
          class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-100  bg-blue-700 text-neutral-0 dark:text-dark-neutral-0 enabled:hover:bg-blue-800 enabled:active:bg-blue-900 dark:bg-blue-400 dark:enabled:hover:bg-blue-300 dark:enabled:active:bg-blue-200 py-2 px-4 relative  "
        >
          <span
            class="inline-flex items-center"
          >
            action
          </span>
        </button>
      </div>
      <div
        class="mr-2 last:mr-0"
      >
        <button
          class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-100  bg-blue-700 text-neutral-0 dark:text-dark-neutral-0 enabled:hover:bg-blue-800 enabled:active:bg-blue-900 dark:bg-blue-400 dark:enabled:hover:bg-blue-300 dark:enabled:active:bg-blue-200 py-2 px-4 relative  "
        >
          <span
            class="inline-flex items-center"
          >
             action
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Banner tests is a snapshot with closeButton equal to true 1`] = `
<div
  class="py-4 pl-4 rounded-[0.1875rem] flex relative bg-blue-100 dark:bg-blue-1000 pr-12 "
  data-testid="test-banner"
>
  <span
    class="w-6 h-6 mr-4"
    role="presentation"
  >
    <svg
      class="fill-blue-700 dark:fill-blue-400 transition-colors duration-150"
      viewBox="0 0 20 20"
    >
      <use
        href="/assets/icons/info.svg#info"
      />
    </svg>
  </span>
  <div>
    <p
      class="text-100 text-neutral-1000 dark:text-dark-neutral-900 transition-colors duration-150"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatibus tempora provident accusamus quaerat iure atque labore odit voluptatem quisquam, maiores voluptas veritatis voluptate sed magni quod enim nobis tempore corrupti.
    </p>
  </div>
  <button
    class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-0  text-blue-700 enabled:hover:text-blue-800 enabled:active:text-blue-900 dark:text-blue-400 dark:enabled:hover:text-blue-300 dark:enabled:active:text-blue-200   text-0 p-2 w-10 h-10 absolute top-2 right-0"
  >
    <span
      class="inline-flex items-center"
    >
      To close
      <svg
        aria-hidden="true"
        class="fill-current w-6 h-6"
        viewBox="0 0 20 20"
      >
        <use
          href="/assets/icons/cancel.svg#cancel"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`Banner tests is a snapshot with hideIcon equal to true 1`] = `
<div
  class="py-4 pl-4 rounded-[0.1875rem] flex relative bg-blue-100 dark:bg-blue-1000 pr-4 "
  data-testid="test-banner"
>
  <div>
    <p
      class="text-100 text-neutral-1000 dark:text-dark-neutral-900 transition-colors duration-150"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatibus tempora provident accusamus quaerat iure atque labore odit voluptatem quisquam, maiores voluptas veritatis voluptate sed magni quod enim nobis tempore corrupti.
    </p>
  </div>
</div>
`;
